connectors/katalog/pkg/connector/credentials.go

```diff
@@ -51,17 +52,30 @@ func (s *DataCredentialsService) GetCredentialsInfo(ctx context.Context, req *co
 		return nil, err
 	}
 
-	// Get the data fields as strings
+	var translationMap = map[string]string{
```

Comment:
What's that? Why are you translating?

Reply:
The secret which contains the dateset credentials contains access_key and access_key keys while the returned Credentials structure expects AccessKey and SecretKey. The modules (copy and arrow-flight) expects access_key and access_key keys and this is why the secret contains those keys. To workaround this mismatch I also had to add access_key and access_key keys to bucket-creds secret which already contains  AccessKey and SecretKey keys.

Reply:
We don't need neither Credentials structure nor the service that returns this structure.

Reply:
@shlomitk1 right but until all the configurations will be ported to use the Vault plugin we should use that...

Reply:
I don't see it. Look at the json tags
https://github.com/IBM/the-mesh-for-data/blob/ab0be89c287a3b0867a95f8d1ffa92fc1195e29a/pkg/connectors/protobuf/credentials.pb.go#L31

Reply:
The following is from the manager log when removing the translation:

`2021-03-24T20:35:00.351Z        INFO    controllers.M4DApplication      Invalid fields in Secret data: json: unknown field "access_key"
2021-03-24T20:35:00.351Z        ERROR   controller      Reconciler error        {"reconcilerGroup": "app.m4d.ibm.com", "reconcilerKind": "M4DApplication", "controller": "m4dapplication", "name": "cba-test-katalog-demo-2", "namespace": "cp4d", "error": "rpc error: code = Unknown desc = Invalid fields in Secret data: json: unknown field \"access_key\""}
github.com/go-logr/zapr.(*zapLogger).Error
        /home/eres/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /home/eres/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:237
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /home/eres/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:209
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
        /home/eres/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:188
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1
        /home/eres/go/pkg/mod/k8s.io/apimachinery@v0.18.6/pkg/util/wait/wait.go:155
`

Reply:
The documentation for katalog say to use accessKey, secretKey, etc. I (mistakenly) used this in the temporary taxonomy instead of using access_key, secret_key, etc. 

Instead of translation make it consistent.
Change from camelCase to snake_case in `/connectors/katalog/manifests/taxonomy.yaml` and regenerate (make generate from katalog dir).

Reply:
Thanks its works find now.